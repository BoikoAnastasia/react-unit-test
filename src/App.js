// import React, { useEffect, useState } from "react";

// const App = () => {
//     const [data, setData] = useState(null);
//     const [toggle, setToggle] = useState(false);
//     const [value, setValue] = useState("");
//     useEffect(() => {
//         setTimeout(() => {
//             setData({});
//         }, 100);
//     }, []);

//     const onClick = () => setToggle(toggle => !toggle)

//     return (
//         <div>
//             <h1 data-testId="input-elem">{value}</h1>
//             {toggle && <div data-testid="toggle-elem">toggle</div>}
//             {data && <div> data</div>}
//             <h1>Hello</h1>
//             <button data-testid="toggle-btn" onClick={onClick}>Click me</button>
//             <input onChange={e=>setValue(e.target.value)} type="text" placeholder="input value" />
//         </div>
//     );
// };

// export default App;

import React from "react";
import AppRouter from "./router/AppRouter";
import Navbar from "./components/Navbar/Navbar";

const App = () => {
    return (
        <div>
            <Navbar/>
            <AppRouter/>
        </div>
    );
};

export default App;
