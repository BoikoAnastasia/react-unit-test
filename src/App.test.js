import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";


describe('test app',()=>{
    test("renders learn react hello", () => {
        render(<App />);
        const helloElement = screen.getByText(/Hello/i);
        const btn = screen.getByRole('button')
        const input = screen.getByPlaceholderText(/input value/i)
        expect(helloElement).toBeInTheDocument();
        expect(btn).toBeInTheDocument();
        expect(input).toBeInTheDocument();
        // screen.debug() 
        expect(input).toMatchSnapshot();      
    });

    test('render any methods', async ()=>{
        render(<App/>)
        const helloQuery = screen.queryByText(/hello2/i)
        expect(helloQuery).toBeNull()

        const helloFind = await screen.findByText(/data/i)
        expect(helloFind).toBeInTheDocument()
    })

    test('render btn', async ()=>{
        render(<App/>)
        const btn = screen.getByTestId('toggle-btn')
        expect(screen.queryByTestId('toggle-elem')).toBeNull()
        fireEvent.click(btn)
        expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument()
        fireEvent.click(btn)
        expect(screen.queryByTestId('toggle-elem')).toBeNull()
    })

    test('render input', async ()=>{
        render(<App/>)
        const input = screen.getByPlaceholderText(/input value/i)
        expect(screen.queryByTestId('input-elem')).toContainHTML('')
        // искуcственное событие
        fireEvent.input(input, {
            target: {value: '123'}
        })
        //близко к пользовательскому событию нажатия клавиш и тд
        userEvent.type(input, '123')
        expect(screen.queryByTestId('input-elem')).toContainHTML('123')
    })
})