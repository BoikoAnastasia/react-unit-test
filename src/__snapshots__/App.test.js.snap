// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app renders learn react hello 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;
